src/main/java/controller/TrackingDataController.java

```diff
@@ -106,6 +112,7 @@ public void loadCSVFile() {
                 newSource.setRepeatMode(true);
                 trackingService.changeTrackingSource(newSource);
                 sourceConnected.setValue(true);
+                visualizationController.setSourceConnected(true);
```

Comment:
We actually have an observer-mechanism in the tracking service to subscribe on connected source events :)
You would then not need to inject the controller in this class. 
But I see that you need it for the visualization, so it's okay to leave it like this for now.